gc/structs/ForwardedHeader.cpp

```diff
@@ -281,7 +281,7 @@ MM_ForwardedHeader::copyOrWaitOutline(omrobjectptr_t destinationObjectPtr)
 
 			if (0 == remainingSizeToCopy) {
 				if (participatingInCopy) {
-					fomrobject_t newValue = ((outstandingCopies - 1) << OUTSTANDING_COPIES_SHIFT) | _beingCopiedTag;
+					fomrobject_t newValue = (fomrobject_t)(((outstandingCopies - 1) << OUTSTANDING_COPIES_SHIFT) | _beingCopiedTag);
```

Comment:
This code still seems wrong but it is wrong for compressed references.  In a compressed references VM an object slot is only 32 bits so performing math on a uintptr_t and then casting it to fomrobject_t is wrong. The `outstandingCopies` variable should be of the appropriate type or should have some sort of mask applied to it before we store it into `newValue`. @amicic what are your thoughts?

Reply:
(fomrobject_t) in this case plays the role of the mask

Reply:
we can put comment now that with (fomrobject_t) downcasting in case of CR, we don't lose significant digits

Reply:
Ok thanks... I am ok with this change but I would like to see the existing issues investigated. Thanks.

Reply:
https://github.com/eclipse/omr/issues/3175 
Please assign to me, and tag appropriately.